modules/search/search-service/main.bicep

```diff
@@ -0,0 +1,101 @@
+metadata name = 'Azure Cognitive Search'
+metadata description = 'Bicep module for simplified deployment of Azure Cognitive Search.'
+metadata owner = 'jong'
+
+@description('Prefix of Azure Cognitive Search Resource Name')
+param prefix string = 'srch'
+
+@description('Name of the Azure Cognitive Search Resource')
+@minLength(3)
+@maxLength(24)
+param name string = take('${prefix}-${uniqueString(resourceGroup().id)}', 24)
+
+@description('Location of the Azure Cognitive Search Resource')
+param location string = resourceGroup().location
+
+@description('Tags of the Azure Cognitive Search Resource')
+param tags object = {}
+
+@description('SKU of the Azure Cognitive Search Resource')
+param sku object = {
+  name: 'standard'
+}
+
+@description('AuthOptions of the Azure Cognitive Search Resource')
+param authOptions object = {}
```

Comment:
It would be nice to create a custom type for the authOptions object: https://learn.microsoft.com/en-us/azure/templates/microsoft.search/searchservices?pivots=deployment-language-arm-template#dataplaneauthoptions-1

Reply:
@shenglol - I added this:

<img width="729" alt="image" src="https://github.com/Azure/bicep-registry-modules/assets/2163001/e6c2df99-52f4-4ec5-b229-ed924232812d">

But I'm getting this error when I run `brm generate`

```
node ➜ .../bicep-registry-modules/modules/search/search-service $ brm generate
[01:48:23 FTL] Unexpected exception.
System.InvalidOperationException: Unable to determine primitive type of null | { aadOrApiKey: aadOrApiKeyType, apiKeyOnly: null | object }
   at Bicep.RegistryModuleTool.ModuleFiles.MainArmTemplateFile.GetPrimitiveTypeName(ITypeReference typeRef)
   at Bicep.RegistryModuleTool.ModuleFiles.MainArmTemplateFile.ToParameter(JsonProperty parameterProperty)
   at System.Linq.Utilities.<>c__DisplayClass2_0`3.<CombineSelectors>b__0(TSource x)
   at System.Linq.Utilities.<>c__DisplayClass2_0`3.<CombineSelectors>b__0(TSource x)
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.UnionIterator`1.GetNext()
   at System.Linq.Enumerable.UnionIterator`1.MoveNext()
   at System.Linq.Enumerable.Aggregate[TSource,TAccumulate](IEnumerable`1 source, TAccumulate seed, Func`3 func)
   at Bicep.RegistryModuleTool.Extensions.EnumerableExtensions.ToMarkdownTable[T](IEnumerable`1 enumerable, Func`2 columnAlignmentSelector)
   at Bicep.RegistryModuleTool.ModuleFiles.ReadmeFile.BuildParametersTable(StringBuilder builder, IEnumerable`1 parameters)
   at Bicep.RegistryModuleTool.ModuleFiles.ReadmeFile.Generate(IFileSystem fileSystem, MainArmTemplateFile mainArmTemplateFile)
   at Bicep.RegistryModuleTool.Commands.GenerateCommand.CommandHandler.<>c__DisplayClass3_0.<InvokeInternal>b__4()
   at Bicep.RegistryModuleTool.Commands.GenerateCommand.CommandHandler.GenerateFileAndLogInformation[T](String fileFriendlyName, Func`1 fileGenerator)
   at Bicep.RegistryModuleTool.Commands.GenerateCommand.CommandHandler.InvokeInternal(InvocationContext context)
   at Bicep.RegistryModuleTool.Commands.BaseCommandHandler.InvokeAsync(InvocationContext context)
```


I have pushed those changes to this PR.  Could you please have a look?

Reply:
Deciding to wait until the next bicep release to change to UDF types.  I kept the code in there commented out to make it easy to make the switch.